[DOCS] Combine search API and URI search API reference docs (#55884)
elastic/elasticsearch: docs/reference/search/search.asciidoc

[[search-search]]
=== Search API
++++
<titleabbrev>Search</titleabbrev>
++++

Returns search hits that match the query defined in the request.

[source,console]
----
GET /twitter/_search
----
// TEST[setup:twitter]

[[search-search-api-request]]
==== {api-request-title}

`GET /<index>/_search`

`GET /_search`

`POST /<index>/_search`

`POST /_search`

[[search-search-api-desc]]
==== {api-description-title}

Allows you to execute a search query and get back search hits that match the
query. You can provide search queries using the <<search-api-query-params-q,`q`
query string parameter>> or <<search-request-body,request body>>.

[[search-search-api-path-params]]
==== {api-path-parms-title}

include::{docdir}/rest-api/common-parms.asciidoc[tag=index]

[[search-search-api-query-params]]
==== {api-query-parms-title}

include::{docdir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
+
Defaults to `true`.

[[search-partial-responses]]
`allow_partial_search_results`::
(Optional, boolean)

> If `true`, partial results are returned for request timeouts one or
> <<shard-failures,partial shard failures>>. If `false`, an error is returned with
> no partial results. Defaults to `true`.

Reviewer: ```suggestion
If `true`, returns partial results if there are request timeouts or
<<shard-failures,shard failures>>. If `false`, returns an error with
no partial results. Defaults to `true`.
```

Author: Thanks for catching this one.

+

> You can override the default for this field by setting
> `search.default_allow_partial_results` to `false` in your cluster settings.

Reviewer: ```suggestion
To override the default for this field, set the 
`search.default_allow_partial_results` cluster setting to `false`.
```
